tfhe/src/high_level_api/integers/oprf.rs

```diff
@@ -0,0 +1,106 @@
+use super::{FheIntId, FheUintId};
+use crate::high_level_api::global_state;
+use crate::high_level_api::keys::InternalServerKey;
+use crate::integer::oprf::SignedRandomizationSpec;
+use crate::{FheInt, FheUint, Seed};
+
+impl<Id: FheUintId> FheUint<Id> {
+    /// Generates an encrypted `num_block` blocks unsigned integer
+    /// taken uniformly in `[0, 2^random_bits_count[` using the given seed
+    /// The encryted value is oblivious to the server
+    /// It can be useful to make server random generation deterministic
+    ///
+    /// ```rust
+    /// use tfhe::integer::gen_keys_radix;
+    /// use tfhe::integer::oprf::SignedRandomizationSpec;
+    /// use tfhe::prelude::FheDecrypt;
+    /// use tfhe::{
+    ///     generate_keys, generate_oblivious_pseudo_random_unsigned_integer, set_server_key,
+    ///     ConfigBuilder, FheUint8, Seed,
+    /// };
+    ///
+    /// let config = ConfigBuilder::default().build();
+    /// let (client_key, server_key) = generate_keys(config);
+    ///
+    /// set_server_key(server_key);
+    ///
+    /// let random_bits_count = 3;
+    ///
+    /// let ct_res: FheUint8 =
+    ///     generate_oblivious_pseudo_random_unsigned_integer(Seed(0), random_bits_count);
+    ///
+    /// let dec_result: u16 = ct_res.decrypt(&client_key);
+    /// assert!(dec_result < (1 << random_bits_count));
+    /// ```
+    pub fn generate_oblivious_pseudo_random_unsigned_integer(
```

Comment:
Since its a associated function of FheUint, no need for `unsigned_integer` in the name